libsolidity/analysis/TypeChecker.cpp

```diff
@@ -2078,6 +2086,75 @@ void TypeChecker::typeCheckABIEncodeFunctions(
 	}
 }
 
+void TypeChecker::typeCheckABIEncodeCallFunction(FunctionCall const& _functionCall)
+{
+	vector<ASTPointer<Expression const>> const& arguments = _functionCall.arguments();
+
+	// Expecting first argument to be the function pointer and second to be a tuple.
+	if (arguments.size() != 2)
+	{
+		m_errorReporter.typeError(
+			6219_error,
+			_functionCall.location(),
+			"Expected two arguments: a function pointer followed by a tuple."
+		);
+		return;
+	}
+
+	auto const functionPointerType = dynamic_cast<FunctionTypePointer>(type(*arguments.front()));
+	auto const argumentTuple = dynamic_cast<TupleType const*>(type(*arguments[1]));
+
+	if (!functionPointerType)
+	{
+		m_errorReporter.typeError(
+			5511_error,
+			_functionCall.location(),
+			"Expected first argument to be a function pointer, not \"" +
+			type(*arguments.front())->canonicalName() +
+			"\"."
+		);
+		return;
+	}
+
+	// There are no checks we can do if all types are accepted
+	if (functionPointerType->takesArbitraryParameters())
+		return;
+
+	std::vector<Type const*> givenArguments;
+
+	// Tuples with only one component become that component
+	if (!argumentTuple)
+		givenArguments.push_back(type(*arguments[1]));
+	else
+		givenArguments = argumentTuple->components();
+
+	size_t const numArguments = functionPointerType->parameterTypes().size();
+
+	if (givenArguments.size() != numArguments)
+	{
+		m_errorReporter.typeError(
+			7788_error,
+			_functionCall.location(),
+			"Expected " + to_string(numArguments) + " instead of " + to_string(givenArguments.size()) + " components for the tuple parameter."
+		);
+		return;
```

Comment:
You could still continue and check the first `min(numArguments, givenArguments.size())` arguments.